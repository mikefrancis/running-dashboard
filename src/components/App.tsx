import * as React from "react";
import DataFetcher from "./DataFetcher";
import MonthlyBreakdown from "./MonthlyBreakdown";
import OverallProgress from "./OverallProgress";

const targetDistance = 500;
export const METRES_PER_KILOMETRE = 1000;

export interface ChartProps {
  data: any[];
  targetDistance: number;
  title: string;
}

const App = () => (
  <div className="bg-black text-white font-sans min-h-screen flex items-center justify-center">
    <DataFetcher>
      {({ data, loading, error }) => {
        if (error) {
          return <span>{error.message}</span>;
        }

        if (loading) {
          return <span>Loading...</span>;
        }

        const chartProps = { data, targetDistance };

        return (
          <>
            <MonthlyBreakdown {...chartProps} title="Breakdown" />
            <OverallProgress {...chartProps} title="Progress" />
          </>
        );
      }}
    </DataFetcher>
  </div>
);

export default App;
