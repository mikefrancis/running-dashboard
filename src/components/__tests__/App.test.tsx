import * as React from "react";
import { render, cleanup, waitForElement } from "react-testing-library";
import axiosMock from "axios";
import App from "../App";

jest.mock("axios");

describe("DataFetcher", () => {
  afterEach(cleanup);

  it("should render its children", () => {
    axiosMock.get.mockResolvedValueOnce({
      data: [
        {
          start_date: "2019-01-01 00:00:00",
          distance: 10000
        }
      ]
    });

    const { container } = render(<App />);

    expect(axiosMock.get).toHaveBeenCalledTimes(1);
    expect(container.firstChild).toMatchSnapshot();
  });
});
