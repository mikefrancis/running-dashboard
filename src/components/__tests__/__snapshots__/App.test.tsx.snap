// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataFetcher should render its children 1`] = `
<div
  class="bg-black text-white font-sans min-h-screen flex items-center justify-center"
>
  <span>
    Loading...
  </span>
</div>
`;
